### PR TITLE
reorganize the return codes and messages a bit. respect http codes

### PR DESCRIPTION
**Pull request by kevinkreiser**
all non http error conditions now return 400 with a reason that varies depending on why the request should not be retried. this also has updates for the messages that come back. it also adds 405 for anything that isnt a GET or POST.

### CODE REVIEW

File: src/loki/service.cc [hunk 1]
```diff
@@ -47,15 +46,21 @@ namespace {
...
+        std::istringstream is(json->second.front());
+        boost::property_tree::read_json(is, pt);
+      }//no json parameter, check the body
+      else if(!request.body.empty()) {
+        std::istringstream is(request.body);
+        boost::property_tree::read_json(is, pt);
+      }
+    }
+    catch(...) {
+      throw std::runtime_error("Failed to parse json request");
```
Comment (kevinkreiser) [inline src/loki/service.cc]: mask whatever boost says when the json is bad with our own message

File: src/loki/service.cc [hunk 2]
```diff
@@ -204,6 +221,7 @@ namespace {
         return result;
       }
       catch(const std::exception& e) {
+        LOG_INFO(e.what());
```
Comment (kevinkreiser) [inline src/loki/service.cc]: log all 400's

File: src/loki/service.cc [hunk 3]
```diff
@@ -177,6 +182,17 @@ namespace {
       try{
         //request parsing
         auto request = http_request_t::from_string(static_cast<const char*>(job.front().data()), job.front().size());
+
+        //block all but get and post
+        if(request.method != method_t::POST || request.method != method_t::GET) {
```
Comment (dgearhart) [inline src/loki/service.cc]: Should be and (&&) not (II)

Comment (kevinkreiser) [inline src/loki/service.cc]: indeed, forthcoming tests i'm writing have exposed this as well! thanks!

**Comment by kdiluca** [comment]
:+1:

**Pull request merged**